src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp

```diff
@@ -0,0 +1,134 @@
+#include <bringauto/fleet_protocol/http_client/FleetApiClient.hpp>
+
+#include <chrono>
+
+using namespace org::openapitools::client;
+
+
+namespace bringauto::fleet_protocol::http_client {
+
+
+FleetApiClient::FleetApiClient(std::string apiUrl, std::string apiKey, std::string companyName, std::string carName,
+                               int32_t maxRequestsThresholdCount, int32_t maxRequestsThresholdPeriodMs,
+                               int32_t delayAfterThresholdReachedMs, int32_t retryRequestsDelayMs) {
+    auto apiConfigPtr = std::make_shared<api::ApiConfiguration>();
+    apiConfigPtr->setBaseUrl(apiUrl);
+    apiConfigPtr->setApiKey("api_key", apiKey);
+    apiClientPtr = std::make_shared<api::ApiClient>();    
+    apiClientPtr->setConfiguration(apiConfigPtr);
+
+    carApi = std::make_shared<api::CarApi>(apiClientPtr);
+    deviceApi = std::make_shared<api::DeviceApi>(apiClientPtr);
+    moduleApi = std::make_shared<api::ModuleApi>(apiClientPtr);
+
+    this->companyName = companyName;
```

Comment:
dont use this-> once correct private names are used, also use ():companyName_(companyName){...} notation

Reply:
changed private names and notation